Source/WebCore/platform/graphics/avfoundation/AudioVideoRendererAVFObjC.mm

```
@@ -0,0 +1,1141 @@
+
+    m_lastSeekTime = seekTime;
+
+    MediaTime synchronizerTime = PAL::toMediaTime([m_synchronizer currentTime]);
+
+    m_isSynchronizerSeeking = std::abs((synchronizerTime - seekTime).toMicroseconds()) > 1000;
+
+    if (!m_isSynchronizerSeeking) {
+        ALWAYS_LOG(LOGIDENTIFIER, "Synchroniser doesn't require seeking current:", synchronizerTime, " seeking:", seekTime);
+        // In cases where the destination seek time precisely matches the synchronizer's existing time
+        // no time jumped notification will be issued. In this case, just notify the MediaPlayer that
+        // the seek completed successfully.
+        m_seekPromise.emplace();
+        Ref promise = m_seekPromise->promise();
+        maybeCompleteSeek();
+        return promise;
+    }
+
+    [m_synchronizer setRate:0 time:PAL::toCMTime(seekTime)];
+
+    // seekState will be Seeking if we first rejected the promise with FlushRequired and flush was subsequently called.
+    if (m_seekState == SeekCompleted || m_seekState == Preparing) {
+        m_seekState = FlushRequired;
+        ALWAYS_LOG(LOGIDENTIFIER, "Requesting Flush");
+        return MediaTimePromise::createAndReject(PlatformMediaError::FlushRequired);
+    }
+
+    m_seekPromise.emplace();
+    return m_seekPromise->promise();
+}
+
+void AudioVideoRendererAVFObjC::setVolume(float volume)
+{
+    m_volume = volume;
+    for (auto& pair : m_audioRenderers) {
+        auto& renderer = pair.second;
+        [renderer setVolume:volume];
+    }
+}
+
+void AudioVideoRendererAVFObjC::setMuted(bool muted)
+{
+    m_muted = muted;
+    for (auto& pair : m_audioRenderers) {
+        auto& renderer = pair.second;
+        [renderer setMuted:muted];
+    }
+}
+
+void AudioVideoRendererAVFObjC::setPreservesPitch(bool preservePitch)
+{
+    m_preservePitch = preservePitch;
+    for (auto& pair : m_audioRenderers) {
+        auto& renderer = pair.second;
+        // False positive see webkit.org/b/298035
```

This comment should be immediately above the line that suppresses the error

---

yeah, bad copy paste here, the line is there twice!